### DISCUSSION TITLE
Fetching models/prices/best model from github instead of more migrations

### DISCUSSION THREAD

**Discussion by drnic** [AllYourBot/hostedgpt#535] [General]
It's another week and there are new openai/anthropic/groq models available and some price changes, etc.

Currently we have `LanguageModel` (per `User`) and I think the only way to programmatically populate this model is via migrations.

<img width="511" alt="image" src="https://github.com/user-attachments/assets/8acfa281-c594-4638-b825-4cfb6552fe82">

I'd propose that we document the available models in this repo, in a `models.json` file, which could then be fetched via HTTP at any time, by any running hostedgpt app to get latest values.

For example, to fetch the raw value of [`fly.toml`](https://github.com/AllYourBot/hostedgpt/blob/main/fly.toml) in the root of the repo, visit <https://raw.githubusercontent.com/AllYourBot/hostedgpt/refs/heads/main/fly.toml>

So a `models.json` would be HTTP retrievable at <https://raw.githubusercontent.com/AllYourBot/hostedgpt/refs/heads/main/models.json>

An airlocked HostedGPT could just load the models for new Users from this file.

An internet-based HostedGPT could fetch the file from the trusted `main` branch from github.

The Language Models page could have a "Sync from Github" button

<img width="1197" alt="image" src="https://github.com/user-attachments/assets/74741fbd-9341-4589-96ec-c9359d207457">

**Comment by keithschacht** [comment]
That’s a great idea. I never considered using the repo to serve a fetchable file like that.

There is one thing we have to keep in mind. Changes to the models.json file are not handled solely with INSERTs. Some existing lines will change, like the default model, and those require UPDATEs to existing records. But that won’t be too tricky the sync code could just iterate through all of them. We have a few PRs where we’ve done this (for the past model changes) so we can use these as a guide.

Is this something you could take the lead on creating a PR for? I can help review and test it out.

**Reply by drnic** [reply]
First draft at https://github.com/AllYourBot/hostedgpt/pull/536?w=1

**Reply by drnic** [reply]
In this PR, it provides tools for insert/update from a local `models.yaml` file; and another rake task to regenerate this file.

Future work could allow this `models.yaml` file to be fetched remotely via a button in the UI.
